frontend/javascripts/dashboard/advanced_dataset/dataset_action_view.tsx

```diff
@@ -37,25 +42,26 @@ function NewAnnotationLink({
   onShowCreateExplorativeModal,
   onCloseCreateExplorativeModal,
 }: {
-  dataset: APIDataset;
+  dataset: APIDatasetCompact;
   isReloading: boolean;
   isCreateExplorativeModalVisible: boolean;
   onShowCreateExplorativeModal: () => void;
   onCloseCreateExplorativeModal: () => void;
 }) {
-  const firstSegmentationLayer = getFirstSegmentationLayer(dataset);
-  const supportsFallback = doesSupportVolumeWithFallback(dataset, firstSegmentationLayer)
-    ? "true"
-    : "false";
-  const fallbackLayerGetParameter =
-    firstSegmentationLayer != null && supportsFallback
-      ? `?fallbackLayerName=${firstSegmentationLayer.name}`
-      : "";
+  // todo: do in backend?
```

Comment:
@fm3 Do you think it would be feasible to have this functionality in the backend? The old code checked whether the dataset has a segmentation layer and generated a URL so that a fallback layer was created for the first segmentation layer. Since this create "button" is a link, the URL needs to exist before clicking on it (otherwise, middle click wouldn't work). The necessary information is not there anymore, though (due the compactness).

I would suggest to have something like `?autoFallback=true` so that the back-end does the functionality explained above.

Reply:
I pushed a commit adding this logic to the backend. Note that `name` is now also optional (as it should follow the name of the segmentation layer if needed). The frontend-sent name has precedence, though.
I adapted admin_rest_api.ts to test this, but it does not yet support all use cases (i.e. the autoFallbackLayer boolean is not yet passed from the react router.

Reply:
excellent :+1: I just incorporated the flag everywhere. However, I noticed that the annotation creation will fail now when autoFallbackLayer is set to true even though there is no volume layer. Since the front-end doesn't know the existing layers, it cannot check this so I would hope the backend can simply ignore the parameter in that case. Does this make sense?